modules/test/playwright/fixtures/featureFlagsTest.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Page, mergeTests, test} from '@playwright/test';
+
+import {loginTest} from './loginTest';
+
+export type FeatureFlagsOptions = {
+	[key: string]: boolean;
+};
+
+export type FeatureFlags = FeatureFlag[];
+
+export interface FeatureFlag {
+	readonly enabled: boolean;
+	readonly key: string;
+}
+
+function featureFlagsTest(options: FeatureFlagsOptions) {
+	const fixtureImpl = test.extend<{
+		featureFlags: FeatureFlags;
+	}>({
+		featureFlags: [
+			async ({page}, use) => {
+
+				// Save original state of FFs
+
+				const originalFeatureFlags: FeatureFlagResult[] = [];
+
+				for (const key of Object.keys(options)) {
+					const result = await isEnabled(key, page);
+
+					originalFeatureFlags.push(result.featureFlag);
+					originalFeatureFlags.push(...result.dependentFeatureFlags);
+				}
+
+				try {
+
+					// Activate  FFs
+
+					for (const [key, enabled] of Object.entries(options)) {
+						await setEnabled(enabled, key, page);
+					}
+
+					// Run test
+
+					await use(
+						Object.entries(options).reduce(
+							(featureFlags: FeatureFlags, [key, enabled]) => {
+								featureFlags.push({
+									enabled,
+									key,
+								});
+
+								return featureFlags;
+							},
+							[]
+						)
+					);
+				}
+				finally {
+
+					// Restore original state of FFs
+
+					originalFeatureFlags.reverse();
+
+					for (const featureFlag of originalFeatureFlags) {
+						await setEnabled(
+							featureFlag.enabled,
+							featureFlag.key,
+							page
+						);
+					}
+				}
+			},
+			{auto: true},
+		],
+	});
+
+	return mergeTests(loginTest, fixtureImpl);
+}
+
+interface IsEnabledResult {
+	dependentFeatureFlags: FeatureFlagResult[];
+	featureFlag: FeatureFlagResult;
+}
+
+interface SetEnabledResult {
+	dependentFeatureFlags: FeatureFlagResult[];
+}
+
+interface FeatureFlagResult {
+	companyId: number;
+	dependenciesFulfilled: boolean;
+	dependencyKeys: string[];
+	description: string;
+	enabled: boolean;
+	featureFlagType: 'BETA' | 'DEPRECATION' | 'DEV' | 'RELEASE';
+	key: string;
+	title: string;
+}
+
+async function isEnabled(key: string, page: Page): Promise<IsEnabledResult> {
+	return invokeServerWithRetry(
+		page,
+		'/o/com-liferay-feature-flag-web/is-enabled',
+		{key}
+	);
```

Comment:
Can we use `Liferay.FeatureFlags['LPS-*']` here? I realize it is not checking the actual state of FF in backend, for example changed by `setEnabled`, but shouldn't UI be reloaded after FF change in any case?

Reply:
Mmm, good idea, I'll test that approach :+1: